layer/executables/entrypoint/dataset.py

```diff
@@ -1,172 +1,67 @@
-    )
-    tracker.mark_running(AssetType.DATASET, dataset_definition.asset_name)
-    # TODO pass path to API instead
-    current_project_uuid = verify_project_exists_and_retrieve_project_id(
-        client, dataset_definition.project_full_name
-    )
-    dataset_build_id = client.data_catalog.initiate_build(
-        current_project_uuid,
-        dataset_definition.asset_name,
-        fabric.value,
-    )
-    dataset_build = client.data_catalog.get_build_by_id(dataset_build_id)
+class DatasetRunner(FunctionRunner):
+    build: DatasetBuild
 
-    if run_id:
-        client.flow_manager.update_run_metadata(
-            run_id=run_id,
-            task_id=dataset_definition.asset_path.path(),
-            task_type=TaskType.DATASET_BUILD,
-            key="build-id",
-            value=str(dataset_build_id),
+    def _create_asset(self) -> Tuple[str, Dict[str, Any]]:
+        self.client.data_catalog.add_dataset(
+            asset_path=self.definition.asset_path,
+            project_id=self.project_id,
+            description=self.definition.description,
+            fabric=self.fabric.value,
+            func_source=self.definition.func_source,
+            entrypoint=self.definition.entrypoint,
+            environment=self.definition.environment,
+        )
+        dataset_build_id = self.client.data_catalog.initiate_build(
+            self.project_id,
+            self.definition.asset_name,
+            self.fabric.value,
         )
 
-    with Context(
-        url=url,
-        client=client,
-        asset_path=dataset_definition.asset_path,
-        dataset_build=dataset_build,
-        tracker=tracker,
-        logged_data_destination=logged_data_destination,
-    ) as ctx:
-        try:
-            with SystemMetrics(logger):
-                result = dataset_definition.func(
-                    *dataset_definition.args, **dataset_definition.kwargs
-                )
-
-            # in case client does not return the dataset
-            if result is None:
-                tracker.mark_done(AssetType.DATASET, dataset_definition.asset_name)
-                client.data_catalog.complete_build(
-                    dataset_build_id,
-                    dataset_definition.asset_name,
-                    dataset_definition.uri,
-                )
-                return
+        self.build = self.client.data_catalog.get_build_by_id(dataset_build_id)
 
-            result = check_and_convert_to_df(result)
-            _run_assertions(
-                dataset_definition.asset_name,
-                result,
-                dataset_definition.assertions,
-                tracker,
+        if self.run_id:
+            self.client.flow_manager.update_run_metadata(
+                run_id=self.run_id,
+                task_id=self.definition.asset_path.path(),
+                task_type=TaskType.DATASET_BUILD,
+                key="build-id",
+                value=str(self.build.id),
```

Comment:
feels like this could also be in common but because all parameters differ almost i understand why it remains here.

this part though is common and important for remote execution, while being not explicit enough at a higher level that we're in a Layer cloud execution environment
```python
if self.run_id:
  self.client.flow_manager.update_run_metadata(
```